src/focus.service.ts

```diff
@@ -509,6 +509,17 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = true;
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
+        visible = false;
+        break;
+      }
+    }
+
+    if (!visible) {
+      return false;
+    }
```

Comment:
Okay, it looks like it doesn't work for elements hidden via `visibility:hidden`. `getComputedStyle` would probably be a decent workaround if you only do it for candidate elements you want to move to.